lib/types.js

```diff
@@ -446,7 +446,15 @@ function getArrayType (typeName, unbox, factory) {
         }
       }
 
-      result.$w = factory.cast(arr, factory.use(typeName), owned);
+      // The type name we get is not always the correct representation of the type so we make it here
+      const internalTypeName = "[L" + elementTypeName.replace(/\./g, '/') + ";";
```

Comment:
For consistency:

```suggestion
      const internalTypeName = '[L' + elementTypeName.replace(/\./g, '/') + ';';
```

(Here and below.)

Could you also check that this passes semistandard? (By running `./node_modules/.bin/semistandard`.)

Reply:
Fixed and rebased, sorry for the delay

Reply:
No worries. Thanks a lot for fixing it! 🙌